scripts/update-component-sha.sh

```diff
@@ -1,10 +1,66 @@
 #!/bin/sh
 set -e
-if [ $# -ne 2 ] ; then
-    echo "Need <OLD> and <NEW> as arguments" >&2
+
+##
+#
+# script to replace hashes in config files
+# see usage() for usage and functionality
+#
+
+function usage() {
+    echo "$0 --<mode> <how-to-find> <new-hash>" >&2
+    echo >&2
+    echo "Avaioable modes: --hash and --image" >&2
+    echo >&2
+    echo "Replace by hash:" >&2
+    echo "	$0 --hash <OLD> <NEW>" >&2
+    echo "	Example: $0 8675309abcdefg abcdef567899" >&2
+    echo "	   Will replace all instances of 8675309abcdefg with abcdef567899" >&2
+    echo >&2
+    echo "Replace by image: $0 --image <IMAGE> <NEW>" >&2
+    echo "	$0 --image <IMAGE> <NEW>" >&2
+    echo "	Example: $0 linuxkit/foo abcdef567899" >&2
+    echo "	   Will tag all instances of linuxkit/foo with abcdef567899" >&2
+    echo >&2
+    echo "By default, for convenience, if no mode is given (--image or --hash), the furst method (--hash) is assumed. Thus the following are equivalent:" >&2
+    echo "	$0 <OLD> <NEW>" >&2
+    echo "	$0 --hash <OLD> <NEW>" >&2
+    echo >&2
+}
+
+
+# backwards compatibility
+if [ $# -eq 2 ]; then
+  set -- "--hash" "$1" "$2"
+fi 
+
+# sufficient arguments
+if [ $# -ne 3 ] ; then
+    usage
     exit 1
 fi
-old=$1
-new=$2
 
-git grep -l "$old" | xargs sed -i -e "s,$old,$new,g"
+
+# which mode?
+case "$1" in
+    --hash)
+        old=$2
+        new=$3
+
+        git grep -l "$old" | xargs sed -i -e "s,$old,$new,g"
+        ;;
+    --image)
+        image=$2
+        hash=$3
+        git grep -l "$image:" | xargs sed -i -e "s,$image:[[:xdigit:]][[:xdigit:]]*,$image:$hash,g"
```

Comment:
Do we have any cases where a given `$image` is a suffix of another? Perhaps use `-E` and `\b` at the start to constrain the match to word boundaries? Oh, or I see `git grep` has `-w` AKA `--word-regexp` which sounds even better (I suppose that applies to my code which became `--hash` mode too).

For the `[[:xdigit:]][[:xdigit:]]` I think POSIX re include counts so `[[:xdigit:]]{40}` should be a safer match (probably not many instances which aren't followed by a character which is not a space or something else not in `[[:xdigit:]]`, but still).